src/main/java/com/amazon/opendistroforelasticsearch/security/transport/DefaultInterClusterRequestEvaluator.java

```diff
@@ -47,15 +56,34 @@
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
 import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
 
-public final class DefaultInterClusterRequestEvaluator implements InterClusterRequestEvaluator {
+public final class DefaultInterClusterRequestEvaluator implements InterClusterRequestEvaluator, DCFListener {
 
     private final Logger log = LogManager.getLogger(this.getClass());
     private final String certOid;
-    private final List<String> nodesDn;
+    private final List<String> staticNodesDnFromEsYml;
+    private boolean dynamicNodesDnConfigEnabled;
+    private volatile Map<String, List<String>> dynamicNodesDn;
 
     public DefaultInterClusterRequestEvaluator(final Settings settings) {
         this.certOid = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CERT_OID, "1.2.3.4.5.5");
-        this.nodesDn = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN, Collections.emptyList());
+        this.staticNodesDnFromEsYml = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN, Collections.emptyList());
+        this.dynamicNodesDnConfigEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN_DYNAMIC_CONFIG_ENABLED, false);
+        this.dynamicNodesDn = Collections.emptyMap();
+    }
+
+    public void subscribeForChanges(DynamicConfigFactory dynamicConfigFactory) {
+        if (this.dynamicNodesDnConfigEnabled) {
+            dynamicConfigFactory.registerDCFListener(this);
+        }
+    }
+
+    private List<String> getNodesDnToEvaluate() {
```

Comment:
what will happen during the node/cluster boot up time when .security_index is not available? dynamicNodesDn will be initialized with empty list?

Reply:
> what will happen during the node/cluster boot up time when .security_index is not available? dynamicNodesDn will be initialized with empty list?

Correct. For boot up it uses the value from YML alone.